File: Solutions/66861/csharp/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace Quera {
    public static class Program {
        public static void Main() {
            var k = GetInputs<int>(1).Single();

            foreach (var goodNumber in GoodNumbers()) {
                if (GetNumOfDivisors(goodNumber) >= k) {
                    Console.WriteLine(goodNumber);
                    break;
                }
            }
        }

        private static IEnumerable<int> GoodNumbers() {
            for (var i = 1; ; i++) 
                yield return (i * (i + 1)) / 2;
        }

        private static int GetNumOfDivisors(int number) {
            var counter = 0;
            for (var i = 1; i <= number; i++) {
                if (number % i == 0)
                    counter++;
            }

            return counter;
        }

        private static List<T> GetInputs<T>(int count) {
            var result = new List<T>(count);
            for (var i = 0; i < count; i++) {
                result.Add((T) Convert.ChangeType(Console.ReadLine(), typeof(T)));
            }

            return result;
        }
    }
}

File: Solutions/66861/python3/program.py
def num_of_divisors(number: int) -> int:
    count = 0
    for i in range(1, number + 1):
        if number % i == 0:
            count += 1
    return count


k = int(input())

n = 1
while True:
    summation = n * (n + 1) // 2
    if num_of_divisors(summation) >= k:
        print(summation)
        break
    n += 1
